boltdesignsystem/bolt
# Keep Drupal Lab Dependencies in Sync + Deploy As Static Site

## Jira

http://vjira2:8080/browse/BDS-1107

## Summary

Create a static version of Drupal Lab and deploy it with our Bolt Design System website.

## Details
Add Tome module to Drupal-lab.
Update Drupal-lab configuration.
Update Tome configuration.
Add NOW.sh routes.
Update Travis configuration.

## How to test
Run this code locally.

### Test steps
- Run `yarn run clean` command on the root folder
- Now run `yarn run setup`
- Go to Drupal-lab folder using command `cd example-integrations/drupal-lab`
- Then run `yarn run generate` to generate Drupal-lab and static version of drupal-lab
- Go back to root folder using `cd ../../` command
- Run `yarn run build`
- Right now run `yarn run serve` to serve bolt
- Go to the provided link in terminal: ` http://localhost:3123`
- Check if website working as expected
- Go to subpage `http://localhost:3123/drupal-lab` to confirm that Drupal-lab website is working (login functionality is not working because is static version)
- Close connection by using `Control+C`

### Develop test
- Run `yarn start` to run docs-site
- Check if development env work correctly

### Travis check
- Go to this [link](https://bugfix-drupal-lab-integration-issues.boltdesignsystem.com/drupal-lab) to check if Drupal-lab static version is correct on Travis build.
